synapse/cortex.py

```diff
@@ -912,8 +890,27 @@ async def finidmon():
 
         await self.postNexsAnit()
 
-        if mirror is not None:
-            await self.initCoreMirror(mirror)
+        if self.mirror is not None:
+            await self._initCoreMirror(self.mirror)
+
+        # Enable leadership change awareness and fire
+        # the leakership hook once at boot
+        self.leaderchangeaware = True
+        await self.iamLeaderHook()
+
+    async def iamLeaderHook(self):
+        if self.nexsroot.amLeader():
+            return await self.doLeaderStuff()
+        return await self.doNonLeaderStuff()
+
+    async def doLeaderStuff(self):
```

Comment:
technically, the login in these could be rolled into `iamLeaderHook` but it does mean that overriding `iamLeaderHook` becomes more complicated